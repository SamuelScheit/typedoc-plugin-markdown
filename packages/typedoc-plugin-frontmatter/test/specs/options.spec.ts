import * as fs from 'fs';
import * as path from 'path';

describe(`Options:`, () => {
  beforeAll(async () => {});

  describe(`YAML`, () => {
    test(`should prepend frontmatter`, async () => {
      const pageContent = fs
        .readFileSync(
          path.join(__dirname, '../out/options/interfaces/SomeInterface.md'),
        )
        .toString();
      expect(pageContent).toMatchSnapshot();
    });

    test(`should prepend frontmatter with preserved tags`, async () => {
      const pageContent = fs
        .readFileSync(
          path.join(__dirname, '../out/options-2/interfaces/SomeInterface.md'),
        )
        .toString();
      expect(pageContent).toMatchSnapshot();
    });
  });
});
