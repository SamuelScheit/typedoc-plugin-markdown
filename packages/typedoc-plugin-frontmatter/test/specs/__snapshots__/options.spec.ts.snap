// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options: YAML should prepend frontmatter 1`] = `
"---
title: SomeInterface
layout: blog
navOrder: 1
hide: true
tagOne: 0
tagTwo: "'Frontmatter' tag"
---

# Interface: SomeInterface
"
`;

exports[`Options: YAML should prepend frontmatter with preserved tags 1`] = `
"---
title: SomeInterface
layout: blog
navOrder: 1
hide: true
tagOne: 0
tagTwo: "'Frontmatter' tag"
---

# Interface: SomeInterface

## Tag One

0

## Tag Two

'Frontmatter' tag
"
`;
